### PR TITLE
roman_datamodels compatibility, move `romandeps` test to CI workflows

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

Tests that rely on the optional Roman dependencies are currently failing since updates to `roman_datamodels` and `rad` that now save ramps as numpy arrays rather than astropy quantities. I've bumped the pin on rdm, and revised how units are handled for that data model in jdaviz. I've also moved the romandeps cron test into `ci_workflows` under allowed failures, so we'll have faster warning next time we're out of sync.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-5016)

### CODE REVIEW

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -72,6 +72,12 @@ jobs:
             toxposargs: --remote-data --run-slow
             allow_failure: true
 
+          - name: Python 3.11 with stable versions of dependencies and Roman
+            python: '3.11'
+            runs-on: ubuntu-latest
+            toxenv: py311-test-romandeps
+            allow_failure: true
```
Comment (bmorris3) [inline .github/workflows/ci_workflows.yml:79]: @pllim Is this the correct transmutation of the [previous cron version](https://github.com/spacetelescope/jdaviz/pull/3353/files#diff-a35c80c05722587c6753fbb714da8527c91b56ed832a898f58b18633dd5af10aL57)?

File: jdaviz/configs/rampviz/plugins/parsers.py [hunk 2]
```diff
@@ -196,6 +196,12 @@ def _roman_3d_to_glue_data(
     data_reshaped = move_group_axis_last(data)
     diff_data_reshaped = move_group_axis_last(diff_data)
 
+    # if the ramp cube has no units, assume DN:
+    if getattr(data_reshaped, 'unit', None) is None:
```
Comment (pllim) [inline jdaviz/configs/rampviz/plugins/parsers.py:200]: Is there a reason to not use `hasattr`?
```suggestion
    if not hasattr(data_reshaped, 'unit'):
```

File: .github/workflows/ci_workflows.yml (hunk 1 above)
Comment (pllim) [reply .github/workflows/ci_workflows.yml:80]: There was a mistake. I pushed commit to hopefully fix it. Does Roman pull in remote data? I think it does, right?

File: jdaviz/configs/rampviz/plugins/parsers.py (hunk 2 above)
Comment (bmorris3) [reply jdaviz/configs/rampviz/plugins/parsers.py:200]: `data_reshaped` could be an `NDData` or `Quantity` with an attr `unit = None`. This approach catches that case.

**Review by pllim — APPROVED** [review]
Thanks!

If your Roman test suite does not use remote data at all (e.g, CRDS or MAST or Box), please remove `toxposargs: --remote-data`. Otherwise, feel free to merge.

**Comment by pllim** [comment]
@bmorris3 are you able to do a manual backport? Thanks!

**Pull request merged**